### PR TITLE
idxmgmt: fix config merging

### PR DESCRIPTION
**Pull request by axw**
## Motivation/summary

Fix a bug related to merging user-defined and default `setup.template` config.

We were merging onto a new config, which meant root config attributes like `path.config` were no longer accessible. The fix is to merge defaults over the root config to ensure the root config attributes are still accessible, and then merge the root config back over that to ensure user-defined config takes precedence.

## Checklist

- [x] Update [CHANGELOG.asciidoc](https://github.com/elastic/apm-server/blob/master/CHANGELOG.asciidoc)

## How to test these changes

See https://github.com/elastic/apm-server/issues/4949

## Related issues

Fixes https://github.com/elastic/apm-server/issues/4949

### CODE REVIEW

**Comment by axw** [comment]
jenkins run the tests please

File: idxmgmt/supporter_factory.go [hunk 1]
```diff
@@ -153,24 +155,21 @@ func (cfg *IndexManagementConfig) logWarnings(log *logp.Logger) {
...
-	return out, nil
+	if configRoot == nil {
+		return defaultConfig, nil
+	}
+	// NOTE(axw) it's important that we merge onto the root config,
+	// due to how config variable resolution works; variables are
+	// resolved using the root of the left-most config in the merge.
+	// We merge the user-defined config back over the defaults to
+	// ensure they take precedence.
+	return common.MergeConfigs(configRoot, defaultConfig, configRoot)
```
Comment (simitt) [inline idxmgmt/supporter_factory.go:174]: This looks complicated - looking at the `common.MergeConfigs` code I couldn't figure out why we need the extra `configRoot`.

Comment (axw) [reply idxmgmt/supporter_factory.go:174]: Agreed, hence the NOTE :)
Is the comment unclear?

**Comment by axw** [comment]
Just found a bug, moving back into draft while I fx.

File: idxmgmt/supporter_factory.go (hunk 1 above)
Comment (axw) [reply idxmgmt/supporter_factory.go:185]: I tweaked the comment slightly, hopefully it's a bit clearer now.

**Review by simitt — APPROVED** [review]

**Pull request merged**